### DISCUSSION TITLE
Recording and release level copyright relationships

### DISCUSSION THREAD

**Discussion by kellnerd** [kellnerd/musicbrainz-scripts#15] [General]
Having a cache for label name to MBID mappings, I only have to add special handling for phonographic copyrights for compilations (which should always be added at release level) before I can switch the script over to batch-add the normal phonographic copyright relationships to all recordings instead of the release.

What do you think, should there be a toggle for choosing between recording and release level relationships?
In which other cases should the phonographic copyright relationships not be added to the recordings?

**Comment by tigerman325** [comment]
I'd also keep phonographic copyright dates for releases as well. I put them on both when possible. This is because I've had digital releases that changed Apple Music or Spotify IDs, etc when the only thing that is different is a new phonographic copyright. But yes, I'd also love to be able to switch to recordings as well. I think the more info on a release helps with identification. The only time I can see them not being added to recordings is compilations. Sometimes, unfortunately, compilations aren't marked as such in the copyright text.

**Reply by kellnerd** [reply]
Ok, so I am going to make it configurable whether the script should also add these to the recording level or only to the release level. Maybe even disable that option by default as this causes less potentially wrong data, experienced editors can always enable it if they know what thery are doing.

**Comment by tigerman325** [comment]
Sounds good.

**Comment by tigerman325** [comment]
Love the new addition of pulling from the annotation!

**Comment by benchwarrant17** [comment]
Me too!

**Comment by tigerman325** [comment]
Is there any way to clear out a wrong label without resetting the entire cache? Example, I had "Sailor Records" as a copyright and used this. However, afterwards, I realized that there were actually 2 Sailor Records, one for an indie label in Denver, one for Steve Miller. I created the new label for Steve Miller as all of his credits using this name had been erroneously added to the indie label. However, anytime I use this, it automatically fills in the indie label because I used it before I realized it was wrong.

**Reply by kellnerd** [reply]
Sure, that's possible:

> In case that an incorrect entity has been selected at some point, the CTRL key allows the user to bypass the cache and force a new search in order to overwrite the old cache entry.

So you can simply parse something like "(C) Sailor Records" with Ctrl+Click and the script will ask you to select a label again. You only have to create the pending relationship to update the cache, but there is no need to actually submit an edit.

**Comment by tigerman325** [comment]
Thank you!!!!! I've been wondering how to do that for a long time. Just never asked.

Marc
________________________________
